src/Automation/Automation/Cmdlet/NewAzureAutomationSchedule.cs

```diff
@@ -149,6 +149,11 @@ public NewAzureAutomationSchedule()
         [PermissionSet(SecurityAction.Demand, Name = "FullTrust")]
         protected override void AutomationProcessRecord()
         {
+            if (!this.IsScheduleNameValid())
+            {
+                throw new ArgumentException(Resources.ScheduleNameInvalid);
```

Comment:
Suggest using common exception type [AzPSArgumentException](https://github.com/Azure/azure-powershell-common/blob/main/src/Common/Exceptions/AzPSArgumentException.cs) for more precise telemetry data
```suggestion
                throw new AzPSArgumentException(Resources.ScheduleNameInvalid, nameof(Name));
```